bindata/v3.11.0/openshift-controller-manager/deploy.yaml

```diff
@@ -0,0 +1,128 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: openshift-controller-manager
+  name: controller-manager
+  labels:
+    app: openshift-controller-manager
+    controller-manager: "true"
+spec:
+  # The number of replicas will be set in code to the number of master nodes.
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 0
+  selector:
+    matchLabels:
+      # Need to vary the app label from that used by the legacy
+      # daemonset ('openshift-controller-manager') to avoid the legacy
+      # daemonset and its replacement deployment trying to try to
+      # manage the same pods.
+      #
+      # It's also necessary to use different labeling to ensure, via
+      # anti-affinity, at most one deployment-managed pod on each
+      # master node. Without label differentiation, anti-affinity
+      # would prevent a deployment-managed pod from running on a node
+      # that was already running a daemonset-managed pod.
+      app: openshift-controller-manager-a
+      controller-manager: "true"
+  template:
+    metadata:
+      name: openshift-controller-manager
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+      labels:
+        app: openshift-controller-manager-a
+        controller-manager: "true"
+    spec:
+      securityContext:
+        runAsNonRoot: true
+        seccompProfile:
+          type: RuntimeDefault
+      priorityClassName: system-node-critical
+      serviceAccountName: openshift-controller-manager-sa
+      containers:
+      - name: controller-manager
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+        image: ${IMAGE}
+        imagePullPolicy: IfNotPresent
+        command: ["openshift-controller-manager", "start"]
+        args:
+        - "--config=/var/run/configmaps/config/config.yaml"
+        resources:
+          requests:
+            memory: 100Mi
+            cpu: 100m
+        ports:
+        - containerPort: 8443
+        terminationMessagePolicy: FallbackToLogsOnError
```

Comment:
I wonder if it would be anyhow useful to have higher `terminationGracePeriodSeconds` than the default `30`. In normal circumstances we probably do not need that